plutus-use-cases/src/Plutus/Contracts/SealedBidAuction.hs

```diff
@@ -0,0 +1,304 @@
+{-# LANGUAGE DataKinds         #-}
```

Comment:
This one is a bit more different, but I'd still be inclined to just modify the original one. These are "demos" of sorts, and I think demoing secret arguments in the main auction example seems fine.

Reply:
English and sealed-bid auctions are different however. Unlike the GameStateMachine example modifying the auction example would make it a different example with a different indended behaviour.

Reply:
Hm. I don't love that we get what looks like an 80% duplicated module and a 100% duplicated test module, though.

Reply:
The test module is not duplicated - in the GameStateModuleWith... example there really is 100% duplication and I've fixed that by just updating the existing example instead (in a commit that I'm just about to push). But I don't think that there is close to 100% duplication between `Spec.Auction` and `Spec.SealedBidAuction`.

Reply:
Yes, you're right, it is somewhat different. It feels like maybe we could share some things, but I think it's fine.